### PR TITLE
add transaction state

### PR DESCRIPTION
**Pull request by tharvik**
* when sending a transaction to self, the account was actually credited with the amount
* adding some tracing calls
* implement `Eq` for `FullTransaction` so we can check that two collections of it are the same
* remove some debug calls in tests
* add transaction state
* simplify `Client::new`
* bump deps

depends on https://github.com/Distributed-EPFL/murmur/pull/14

### CODE REVIEW

File: tests/send-asset-to-itself-keep-balance [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+source ./lib.sh
+
+start_network
+
+itself=$(client config new $(get_node_rpc))
+itself_pubkey=$(echo "$itself" | client config get-public-key)
```
Comment (ineiti) [inline tests/send-asset-to-itself-keep-balance:8]: Just a random thought - nothing to change: what if the configuration of `client config new` would be a JSON, then you could use `jq` to parse it?

**Review by ineiti — APPROVED** [review]
Except for missing unit-tests, this looks good to me.

File: tests/send-asset-to-itself-keep-balance (hunk 1 above)
Comment (tharvik) [reply tests/send-asset-to-itself-keep-balance:8]: Hum, TOML and JSON are equivalent, do you know a tool that translate between the two maybe?
I'm in favor of keeping TOML as it's a "human" format, and providing small cmds to extract needed information (config edit as CLI in short).

**Comment by tharvik** [comment]
> Except for missing unit-tests

Added some :)

File: tests/send-asset-to-itself-keep-balance (hunk 1 above)
Comment (ineiti) [reply tests/send-asset-to-itself-keep-balance:8]: So after @cgrigis presentation we should not be using these anyway but using environment variables ;) And btw, there is a new kid on the block by Amazon:

https://amzn.github.io/ion-docs/

Just for time travellers: this is chit-chat and has nothing to do with the actual code here. Just thinking out loud...

Comment (tharvik) [reply tests/send-asset-to-itself-keep-balance:8]: Yep, I'm thinking on how to use env var to do that, such as how to specify the network of nodes for the server (`NODES='$pubkey_1@$address_1 $pubkey_2@$address_2 ...'`?). I'm not sure I agree a 100% with using only env variables, it fits for many cases but maybe not all.

Hoo, `ion` looks interesting, a human readable protobuf maybe?

File: src/bin/server/accounts/mod.rs [hunk 2]
```diff
@@ -208,7 +208,67 @@ impl AccountsHandler {
...
+            accounts
+                .get_last_sequence(user_pubkey)
+                .await
+                .expect("to get last sequence"),
+            new_account.last_sequence(),
+        );
+    }
+
+    #[tokio::test]
+    async fn transfer_to_themself_increment_sequence_and_keep_balance() {
```
Comment (ineiti) [inline src/bin/server/accounts/mod.rs:244]: https://www.justice.gc.ca/eng/rp-pr/csj-sjc/legis-redact/legistics/p1p30.html

File: tests/send-asset-to-itself-keep-balance (hunk 1 above)
Comment (cgrigis) [reply tests/send-asset-to-itself-keep-balance:8]: Like all rules, I would suggest to not take them as "absolute" but evaluate when/whether they really make sense for the task at hand ;-)

**Pull request merged**